scenario_player/utils/legacy.py

```diff
@@ -197,35 +197,36 @@ def _set_timeout(self, timeout=None):
             self._endtime = time.time() + timeout
 
 
-def wait_for_txs(
-    client_or_web3: Union[Web3, JSONRPCClient], txhashes: Set[TransactionHash], timeout: int = 360
-):
-    if isinstance(client_or_web3, Web3):
-        web3 = client_or_web3
-    else:
-        web3 = client_or_web3.web3
+def wait_for_txs(client: JSONRPCClient, txhashes: Set[TransactionHash], timeout: int = 360):
+
     start = time.monotonic()
     outstanding = False
-    txhashes = set(txhashes)
-    while txhashes and time.monotonic() - start < timeout:
+    txhashes_bytes = []
+
+    ## for some reason txhashes is a set of str, not bytes
+    for txhash in txhashes:
+        if isinstance(txhash, str):
+            txhashes_bytes.append(bytes.fromhex(remove_0x_prefix(txhash)))
```

Comment:
As discussed in person we should fix this at the source.

Reply:
done, I now fixed that at the source directly to convert any str into bytes

Reply:
this will be covered here https://github.com/raiden-network/scenario-player/issues/476